src/cluster.rs

```diff
@@ -491,10 +495,8 @@ impl ClusterConnection {
                         if kind == ErrorKind::Ask {
                             asking = err.redirect_node().map(|x| x.0.to_string());
                         } else if kind == ErrorKind::Moved {
-                            // Refresh slots and request again.
-                            self.refresh_slots()?;
-                            excludes.clear();
-                            continue;
+                            // follow the Moved hint from Redis
+                            moved = err.redirect_node().map(|x| x.0.to_string());
```

Comment:
`redirect_node` does explicitly check for `ErrorKind::Ask` and will return `None` if it isn't.
This usage here is thus essentially a no-op.

Reply:
OK. I'll do some more testing and see if I can isolate why this did not work for us. I'm definitely not ruling out that we made an implementation error! 😊